brave/src/main/java/brave/handler/MutableSpan.java

```diff
@@ -746,6 +748,129 @@ static boolean updateOrRemove(ArrayList list, int i, Object value, @Nullable Obj
       && equal(error, that.error);
   }
 
+  /** Writes this span in Zipkin V2 format */
+  // Ported from zipkin2.internal.V2SpanWriter and may eventually move to a separate codec type
+  @Override public String toString() {
+    StringBuilder b = new StringBuilder();
```

Comment:
Forget if we have a ThreadLocal StringBuilder buffer, it would probably help here. But can do it later if we don't have it conveniently available.

Reply:
if we bound it I suppose we could? This is currently only used in logging and I'm not sure if we want to leave a stringbuilder on each thread just in case toString is called...

Reply:
in a reporter totally agree (as the reporter thread is likely only 1 or few)

Reply:
by logging I mean if no-one sets a handler or a spanReporter

Reply:
Ah yeah that sounds good